core/commands/p2p.go

```diff
@@ -296,20 +313,19 @@ var p2pCloseCmd = &cmds.Command{
 		}
 
 		match := func(listener p2p.Listener) bool {
-			out := true
-
-			if p {
-				out = out && (proto == listener.Protocol())
+			if closeAll {
+				return true
 			}
-			if l {
-				out = out && (listen == listener.ListenAddress())
+			if p && proto != listener.Protocol() {
+				return false
 			}
-			if t {
-				out = out && (target == listener.TargetAddress())
+			if l && listen != listener.ListenAddress() {
```

Comment:
Ah. So, I wasn't paying attention... this needs to be `listen.Equals(listener.ListenAddress())` (same with target below). They're not strings.